pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -240,6 +247,25 @@ func (c *Cluster) enablePrometheusModule(clusterName string) error {
 	return nil
 }
 
+func (c *Cluster) enableMgrModules(context *clusterd.Context, clusterName string, mgrSpec cephv1.MgrSpec) error {
+	// Enable mgr modules from the spec
+	for _, module := range mgrSpec.Modules {
+		if err := client.MgrEnableModule(c.context, c.Namespace, module.Name, true); err != nil {
+			return fmt.Errorf("failed to enable mgr %s module. %+v", module, err)
+		}
+		// Ceph Octopus will have that option enabled
+		if module.Name == "pg_autoscaler" && c.clusterInfo.CephVersion.IsAtLeastNautilus() {
+			monStore := config.GetMonStore(context, clusterName)
+			err := monStore.Set("global", "osd_pool_default_pg_autoscale_mode", "on")
```

Comment:
This will re-enable the autoscale module with every orchestration, even if the user has configured this to be `off`. I thought we talked about how this would be a bad idea.

Reply:
I thought we discussed it would be a bad idea if we were always doing this. In this PR we are only doing this when `pg_autoscaler` is in the list of modules to enable in the CR, the user should expect autoscaling to be enabled, right? If they don't want the autoscaler enabled, then they wouldn't add it to the CR.